### ISSUE TITLE
Conv2d layers should accept 4-tuple for padding argument

### ISSUE THREAD

**Issue by sytelus** [pytorch/pytorch#17150]
## 🚀 Feature
Currently Conv2d layers supports int or tuple however tuple is only allowed to be 2-tuple and so this only supports symmetrical padding. This should be 4-tuple to allow for asymmetrical padding.

## Pitch and Motivation

One of the first issue users coming from TensorFlow encounter is lack of feature `padding='SAME'`. This feature has allowed users to quickly prototype without having to dive into convolutional arithmetic for network design. There have been [reservations](https://github.com/pytorch/pytorch/issues/3867) for adding this feature despite of very strong demand from users. I think the list that can be done is above simple change that won't cause any other issues.

## Alternatives

Current alternative for users is to spend time in learning and extracting formulas for convolutional arithmetic from Internet sources, neccessory padding and use either `F.pad` or padding layers (which are unfortunately not easily discoverable).

**Comment by sytelus** [comment]
I think this should be promoted to bug because of the following finding:

For `ConvTranspose2d`, the padding decreases output shape (as opposed to increase for `Conv2D`. However as both only allow even value for total padding (specified padding is applied on both sides), this means there is no way to achieve symmetrical autoencoder-decoder architectures. 

So in nutshell, the frameworks that don't allow specifieng asymmetrical padding are incomplete and cannot be used to replicate architectures that something like Tensorflow allows without compromising in layer shapes.

I feel this is very fundamental limitation. Please let me know if this is unclear.

**Comment by soumith** [comment]
@sytelus we have `output_padding` in `ConvTranspose2d` to have a right padding / right rounding that makes sure we can create symmetric architectures.

**Comment by leimao** [comment]
Looking at this in 2021, any updates?
